test/playwright.fixtures.ts

```diff
@@ -182,20 +182,29 @@ registerWorkerFixture('golden', async ({browserName}, test) => {
   await test(p => path.join(browserName, p));
 });
 
-registerFixture('context', async ({browser}, test) => {
+registerFixture('context', async ({browser, toImpl}, runTest, config, test) => {
   const context = await browser.newContext();
-  await test(context);
+  if (toImpl) {
+    const traceStorageDir = path.join(config.outputDir, 'trace-storage');
+    const relativePath = path.relative(config.testDir, test.file).replace(/\.spec\.[jt]s/, '');
+    const sanitizedTitle = test.title.replace(/[^\w\d]+/g, '_');
+    const traceFile = path.join(config.outputDir, relativePath, sanitizedTitle + '.trace');
+    toImpl(context)._traceTo(traceStorageDir, traceFile);
+  }
+  await runTest(context);
   await context.close();
 });
 
-registerFixture('page', async ({context}, runTest, config, test) => {
+registerFixture('page', async ({context, toImpl}, runTest, config, test) => {
   const page = await context.newPage();
   await runTest(page);
   if (test.error) {
     const relativePath = path.relative(config.testDir, test.file).replace(/\.spec\.[jt]s/, '');
     const sanitizedTitle = test.title.replace(/[^\w\d]+/g, '_');
     const assetPath = path.join(config.outputDir, relativePath, sanitizedTitle) + '-failed.png';
     await page.screenshot({ path: assetPath });
+    if (toImpl)
+      await toImpl(page)._captureSnapshot({ timeout: 5000, label: 'Test Failed' });
```

Comment:
Let's make it available for real

Reply:
I'll do that once we are ready for some kind of api :)